ipam/allocator.go

```diff
@@ -115,6 +117,41 @@ func (a *Allocator) updateBitMasks(aSpace *addrSpace) error {
 	return nil
 }
 
+// Fixes damaged bitmask. Meant to be called in constructor only.
+// Check could be positive during docker upgrade from 1.9.1 to newer versions
+func (a *Allocator) checkConsistency(as string) {
+	// List of affected master pools
+	pdList := map[string]*PoolData{}
+
+	// Retrieve current configuration and bitmask data from
+	// the datastore about this address space
+	a.refresh(as)
+	aSpace := a.addrSpaces[as]
+	a.updateBitMasks(aSpace)
+
+	// Look for corrupted bitmask data
+	for sk, pd := range aSpace.subnets {
+		if pd.Range != nil {
+			continue
+		}
+		bm := a.addresses[sk]
+		if bm.IsCorrupted() {
+			log.Debugf("Detected corrupted bitmask: %s. Fixing it.", bm)
+			pdList[sk.String()] = pd
+		}
+	}
+
+	// Re install the affected address pools
+	for sk, pd := range pdList {
+		if err := a.ReleasePool(sk); err != nil {
```

Comment:
Unfortunately it is not possible to reconstruct all the info from a damaged bitmask being the bit sequence the source of truth.

For the global scope case, the consistency check and fix can only come from the controller, which has access to the endpoints data.

It should somehow lock all the API, perform the reconciliation and unlock.